tests/src/Flyvemdm/Tests/CommonTestCase.php

```diff
@@ -39,25 +40,38 @@ class CommonTestCase extends GlpiCommonTestCase {
     * the agent returned will not contain an ID. To ensore the enrollment succeeded
     * use isNewItem() method on the returned object.
     *
-    * @param \User $user
+    * @param integer $userId
     * @param array $input enrollment data for agent creation
+    * @param boolean $keepLoggedUser allow change of user for testing
     * @return \PluginFlyvemdmAgent
     */
-   protected function enrollFromInvitation(\User $user, array $input) {
+   protected function enrollFromInvitation($userId, array $input, $keepLoggedUser = true) {
       // Close current session
+      $currentUser = $_SESSION['glpiname'];
       $this->terminateSession();
       $this->restartSession();
       $this->setupGLPIFramework();
 
       // login as invited user
-      $_REQUEST['user_token'] = \User::getToken($user->getID(), 'api_token');
+      $_REQUEST['user_token'] = \User::getToken($userId, 'api_token');
       $this->boolean($this->login('', '', false))->isTrue();
-      $this->setupGLPIFramework();
       unset($_REQUEST['user_token']);
 
       // Try to enroll
       $agent = new \PluginFlyvemdmAgent();
       $agent->add($input);
+      if (!$agent->isNewItem()) {
+         // item has been created but its data is incomplete so let's load it
+         $agent->getFromDB($agent->getID());
+      }
+
+      if ($keepLoggedUser && $currentUser) {
+         // Go back to previous logged user
+         $this->terminateSession();
+         $this->restartSession();
+         $this->setupGLPIFramework();
+         $this->login($currentUser, $currentUser);
```

Comment:
this will not work with all user accounts.

Reply:
I know that but we can't now the password and also the test are using glpi user for login right now, We can set the same username and password for our test.

Reply:
someday we may need other use cases. We may add a test to check the login succeeded. This way we immediately know why subsequent execution flow fails, if it happens.

Reply:
I added one asserter for that case